python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -282,14 +281,15 @@ def get_asset(
 
     check.inst_param(asset_key, "asset_key", AssetKey)
     instance = graphene_info.context.instance
-    remote_node = graphene_info.context.get_asset_node(asset_key)
 
-    if not remote_node and not instance.has_asset_key(asset_key):
+    has_remote_node = graphene_info.context.asset_graph.has(asset_key)
+    if not has_remote_node and not instance.has_asset_key(asset_key):
         return GrapheneAssetNotFoundError(asset_key=asset_key)
-    elif remote_node:
+
+    if has_remote_node:
         def_node = _graphene_asset_node(
             graphene_info,
-            remote_node,
+            graphene_info.context.asset_graph.get(asset_key),
```

Comment:
remember this coming up on the other PRs too - is it useful to separate out has() from get() != null?